app/views/fenrir_view/styleguide/preview.html.erb

```diff
@@ -0,0 +1,53 @@
+<h2><%= @component.name.titleize %></h2>
+
+<% @component.stub_properties.each_with_index do |property, property_index| %>
+  <div class="fenrir-view-component-example__examples-wrapper">
+    <% @component.devices.each_with_index do |device, device_index| %>
+        <iframe id="<%= @component.name%>-<%= property_index %>-<%= device_index %>" width="<%= device.width %>" height="<%= device.height %>" name="<%= device.name %>" srcdoc='
+          <script type="text/javascript">
+            window.onload = function() {
+              if (parent) {
+                var oHead = document.getElementsByTagName("head")[0];
+                var arrStyleSheets = parent.document.getElementsByTagName("link");
+                var arrJavaScript = parent.document.getElementsByTagName("script");
+
+                for (var i = 0; i < arrStyleSheets.length; i++) {
+                  oHead.appendChild(arrStyleSheets[i].cloneNode(true));
+                }
+
+                for (var i = 0; i < arrJavaScript.length; i++) {
+                  var script = document.createElement("script");
+                  script.type = "text/javascript";
+
+                  if (arrJavaScript[i].src) {
+                    script.src = arrJavaScript[i].src;
+                    oHead.appendChild(script);
+                  }
+                }
+
+                document.body.classList.add("fenrir-view-component-example__content-body")
+              }
+            }
+          </script>
+          <div>
+              <%= ui_component( @component.name, property.fetch(:properties)) do |layout| %>
+                <% @component.yields(property.fetch(:yields)).each do |content| %>
+                  <% if content.key? && content.args.any? %>
+                    <% layout.public_send(content.key, content.args) do %>
+                      <%= content.content %>
+                    <% end %>
+                  <% elsif content.key? %>
+                    <% layout.public_send(content.key) do %>
+                      <%= content.content %>
+                    <% end %>
+                  <% else %>
+                    <%= content.content %>
+                  <% end %>
+                <% end%>
+              <% end %>
+          </div>
+        '>
+        </iframe>
+    <% end %>
+  </div>
```

Comment:
I'm not sure we need the iframes or multiple device size rendering of the components for the accessibility testing?

There might be components where it would be nice to test both the mobile and desktop view, but I guess we could do that by resizing the viewport instead of rendering both :)

I think we can simplify this page, and maybe make it more stable too by not relying on each iframe to separately load in the CSS from the view with css on load :)


```suggestion
<% @component.stub_properties.each_with_index do |property, property_index| %>
  <%= ui_component( @component.name, property.fetch(:properties)) do |layout| %>
    <% @component.yields(property.fetch(:yields)).each do |content| %>
      <% if content.key? && content.args.any? %>
        <% layout.public_send(content.key, content.args) do %>
          <%= content.content %>
        <% end %>
      <% elsif content.key? %>
        <% layout.public_send(content.key) do %>
          <%= content.content %>
        <% end %>
      <% else %>
        <%= content.content %>
      <% end %>
    <% end%>
  <% end %>
<% end %>
```

Reply:
yeah makes sense, I'll update the view and remove the iframes !